packages/teams-js/src/internal/utils.ts

```diff
@@ -518,3 +518,49 @@ const supportsPerformanceTimers = !!performance && 'now' in performance;
 export function getCurrentTimestamp(): number | undefined {
   return supportsPerformanceTimers ? performance.now() + performance.timeOrigin : undefined;
 }
+
+/**
+ * @hidden
+ * @internal
+ * Limited to Microsoft-internal use
+ *
+ * Function to check whether the data is a primitive type or a plain object.
+ * @param value The value to check
+ * @returns true if the value is a primitive type or a plain object, false otherwise
+ *
+ */
+export function isPrimitiveOrPlainObject(value: unknown, depth: number = 0): boolean {
+  if (depth > 1000) {
```

Comment:
I added this change as well since you last reviewed @TrevorJoelHarris

Reply:
Because Anand from 3pcxp said, he does not see it going any further than that

Reply:
nit: It would be great if you could add this context as a comment for this function to help the next developer understand the logic.

Reply:
I added a comment saying `Recursion is limited to a maximum depth of 1000 to prevent excessive nesting and potential stack overflow.` Does that work?